### DISCUSSION TITLE
Detect if client canceled the request

### DISCUSSION THREAD

**Discussion by lsmith77** [fastapi/fastapi#8699] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post("/foo", status_code=201)
def foo(
    request: Request, user_request_in: RequestInEvent, background_tasks: BackgroundTasks
):
    .. do some work ..

    background_tasks.add_task(write_user_training_data, request, user_request_in)

    return ..
```

### Description

I have an API that sends an API request for every key pressed in a client. Since its quite common that another key is pressed before the request returns, the client cancels these requests.

Is there some way I can detect if the request was canceled inside the background task?

### Operating System

Linux

### FastAPI Version

0.68.2

### Python Version

3.9

**Comment by raphaelauv** [comment]
fastAPI is not a workload management process. Also your question is already on stackoverflow -> https://stackoverflow.com/questions/68300429/fastapi-background-task-kill

I think you can close this issue @lsmith77  , Thank you

**Comment by lsmith77** [comment]
It seems my explanation was not clear. 

I am not looking for FastAPI to automatically end execution of the background task if the client cancels the request.

I want the background task to run, but inside the background task I want to be able to log if the client actually read the response or not.
